spec/system/composer/prosemirror_editor_spec.rb

```diff
@@ -23,12 +23,20 @@ def open_composer
   end
 
   def paste_and_click_image
+    # This helper can only be used reliably to paste a single image when no other images are present.
+    expect(rich).to have_no_css(".composer-image-node img")
+
     cdp.allow_clipboard
     cdp.copy_test_image
     cdp.paste
-    # account for the debounced resolution
-    sleep 0.2
+
+    expect(rich).to have_css(".composer-image-node img", count: 1)
+    expect(rich).to have_no_css(".composer-image-node img[src='/images/transparent.png']")
+    expect(rich).to have_no_css(".composer-image-node img[data-placeholder='true']")
```

Comment:
I'm not wild about this being the way to detect if the image is ready to be clicked, but I haven't seen neater method.